core/src/main/java/io/kestra/core/repositories/FlowRepositoryInterface.java

```diff
@@ -38,6 +38,15 @@ default Flow findByExecution(Execution execution) {
         }
     }
 
+    default Optional<Flow> findOptionalByExecution(Execution execution) {
```

Comment:
Not a fan of the name of the method.
You can either update findByExecution to return an optional (and impact all usage) or use the findById() in the only place that needs this method. But adding a method in the interface each time you need something slightly different is a code smell.